src/gui/properties/speedplotview.cpp

```diff
@@ -169,6 +170,67 @@ int SpeedPlotView::maxYValue()
     return maxYValue;
 }
 
+namespace
+{
+    // table of supposed nice steps for grid marks to get nice looking quarters of scale
+    const double roundingTable[] = {1.2, 1.6, 2, 2.4, 2.8, 3.2, 4, 6, 8, 10};
+    qint64 getNiceYScale(double value)
+    {
+        if ( value == 0.0  ) return 0;
+        if ( value <= 12.0 ) return 12;
+        qint64 scale = 1;
+        while (value > 1024) {
+            value /= 1024;
+            scale *= 1024;
+        }
```

Comment:
In all this cycle variants there are 1 comparison and 2 arithmetical operations in cycle body. And in your variants also adds 1 operation after exit of the cycle. Let's cout operations on starting value=2000 for example.
1. My variant.  
```
//condition calculation
- comparison
//body
- operation
- operation
//condition calculation
- comparison
//cycle exit
```

Total is 2 comparisons and 2 operations.

2. Your variant (both of them are identical in code structure).
```
//condition calculation
- operation
- comparison
//body
- operation
//condition calculation
- operation
- comparison
//cycle exit
- operation
```
Total is 2 comparisons and 4 operations.

What about replacing arithmetic with shifts - modern compilers will do it itself, if it really gives some gain. And the age of 8 bit CPU when it gives significant profit has gone log time ago.
Anyway, this is not a hot spot code, where we should fight for every cycle. So really only maintainability considerations are applicable IMHO.

Reply:
I just hate to see such suboptimal calculations in a loop. But here we will usually have 2-3 steps, so it is not so important.

BTW, it can be optimized even more than in my example.